controllers/storagecluster/reconcile.go

```diff
@@ -112,13 +112,13 @@ var validTopologyLabelKeys = []string{
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *StorageClusterReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.Log = r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 
```

Comment:
Please hold onto the old logger and restore it at the end of the function using a defer func. We don't want to be logging with the stale namespace and name outside of the reconcile function.

Reply:
Done